In [1]:
import pandas as pd

In [2]:
books = pd.read_csv("allbacks.csv")
books.head()

,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb


In [3]:
books["cover"].sort_values().reset_index(drop=True)

0     hb
1     hb
2     hb
3     hb
4     hb
5     hb
6     hb
7     pb
8     pb
9     pb
10    pb
11    pb
12    pb
13    pb
14    pb
Name: cover, dtype: str

the Median in (pb) which is the value of number 8th

In [4]:
sorted_covers = books["cover"].sort_values().reset_index(drop=True)
median_cover = sorted_covers.iloc[len(sorted_covers) // 2]

print("Median cover type:", median_cover)

Median cover type: pb


In [5]:
mean_weight = books["weight"].mean()
print("Mean weight of all books:", mean_weight)

Mean weight of all books: 706.6666666666666


The mean weight of all books is 706.67 grams

In [6]:
mean_by_cover = books.groupby("cover")["weight"].mean()
print(mean_by_cover)

cover
hb    796.428571
pb    628.125000
Name: weight, dtype: float64



The mean of hb weighs about 796.43 grams, while the meanof pb weighs about 628.13 grams.

The hardcover books are about 168.30 grams heavier than the softcover, which suggests that the cover type is associated with a difference in book weight in this inventory.

In [7]:
print("Median cover type:", median_cover)
print("Mean weight of all books:", mean_weight)
print("Mean hardcover weight:", mean_by_cover["hb"])
print("Mean paperback weight:", mean_by_cover["pb"])

Median cover type: pb
Mean weight of all books: 706.6666666666666
Mean hardcover weight: 796.4285714285714
Mean paperback weight: 628.125
